python/vertx/eventbus.py

```diff
@@ -0,0 +1,346 @@
+                    self._state = _State.CLOSED
+                    break
+                json_message = payload.decode('utf-8')
+                message = json.loads(json_message)
+                if message['type'] == 'message':  # message
+                    if 'address' not in message:
+                        self._err_handler(message)
+                    else:
+                        if message['address'] in self.handlers:
+                            for handler in self.handlers[message['address']].all_handlers():
+                                handler(message)
+                        else:
+                            LOGGER.warning("No handler found on address %s" % message['address'])
+                elif message['type'] == 'err':  # err
+                    self._err_handler(message)
+                elif message['type'] == 'pong':  # ping/pong
+                    self.last_pong = time.time()
+                    LOGGER.debug("get pong response")
+                else:  # unknown message type
+                    self._err_handler(message)
+            except socket.timeout:
+                LOGGER.debug("timeout, try again")
+                continue
+            except Exception as e:
+                if self._state == _State.CLOSED:
+                    LOGGER.debug("client has been closed")
+                else:
+                    if e.args[0] == errno.ECONNRESET:
+                        self._state = _State.CLOSED
+                        self.handlers.clear()
+                        LOGGER.debug("connection reset by server")
+                    else:
+                        self._state = _State.BROKEN
+                        LOGGER.error("Connection was broken", exc_info=e)
+                break
+        if self.auto_connect and self._state != _State.CLOSED:
+            self.connect()
+
+    def is_connected(self):
+        return self._state == _State.CONNECTED
+
+    def close(self):
+        if self._state != _State.CLOSED:
+            try:
+                self._state = _State.CLOSING
+                self.sock.close()
+                self._state = _State.CLOSED
+                self.handlers.clear()
+            except Exception as e:
+                LOGGER.error("Failed to close the socket", exc_info=e)
+
+    def _check_closed(self):
+        if not self.is_connected():
+            if self.auto_connect and self._state != _State.CLOSED:
+                self.connect()
+            else:
+                raise Exception("socket has been closed.")
+
+    # send, receive, register, unregister ------------------------------------
+
+    def _send_frame(self, message_s):
+        message = message_s.encode('utf-8')
+        frame = struct.pack('!I', len(message)) + message
+        self.sock.sendall(frame)
+
+    def send(self, address, headers=None, body=None, reply_address=None, reply_handler=None):
+        self._check_closed()
+        ra = reply_address
+        rh = reply_handler
```

Comment:
These variable names are too short. Why not just use the original method parameters and modify `reply_address` when it is `None`?
While modifying parameters can make code difficult to read, if it's just to handle a default value, it's OK.